hudi-client/hudi-client-common/src/main/java/org/apache/hudi/client/transaction/MetadataTableNonBlockingWritesConflictResolutionStrategy.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.hudi.client.transaction;
+
+import org.apache.hudi.common.model.HoodieCommitMetadata;
+import org.apache.hudi.common.table.HoodieTableMetaClient;
+import org.apache.hudi.common.table.timeline.HoodieInstant;
+import org.apache.hudi.common.util.Option;
+import org.apache.hudi.exception.HoodieMetadataException;
+import org.apache.hudi.exception.HoodieWriteConflictException;
+import org.apache.hudi.table.HoodieTable;
+
+import java.util.stream.Stream;
+
+/**
+ * Conflict resolution strategy to be used with metadata table when NBCC is enabled.
+ * This resolution will not conflict any operations and let everything succeed the conflict resolution step.
+ */
+public class MetadataTableNonBlockingWritesConflictResolutionStrategy implements ConflictResolutionStrategy {
```

Comment:
One thing to note for subsequent PRs adding the streaming write support on MDT: the requested time ordering can be different from completion time ordering in MDT after DT and MDT start transactions together (and the completion time order is supposed to be the same between MDT and DT, because of the DT lock).  In this case, when reading MDT file groups, we need to make sure that the completion time ordering is used for merging log files.

```
   MDT DC 1 |-----------------|
   MDT DC 2      |---------|
```

Reply:
yeah, I have not found a right place yet to add this as java docs. will try to add this in future patches.